### PR TITLE
Fix/members owner leave team action

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

Members who are owners can now leave their own organization.
**but** only if 1 other owner is in the organization, so the org does not become an orphan.

## What is the current behavior?

Owners cannot leave their own org, even with multiple other owners.
Although Owners can delete other owners.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Also removed the old 'transfer ownership' actions and code

### CODE REVIEW

**Comment by MildTomato** [comment]
latest docs update 
https://docs-git-fix-members-owner-leave-team-action-supabase.vercel.app/docs/guides/platform/access-control#transferring-ownership-of-an-organization

File: apps/docs/pages/guides/platform/access-control.mdx [hunk 1]
```diff
@@ -37,6 +37,15 @@ You can invite your team members into your organizations to collaborate on proje
...
 </video>
 
+### Transferring ownership of an organization
+
+An organization can have multiple owners, so a Supabase organization can either have 1 owner, or several owners.
+
+If you would like to no longer be an owner of an organization you can simply click 'Leave team' in the members view (`https://app.supabase.com/org/<org-slug>/settings#team`) of your organization.
+However, you can only leave an organization when there is _at least 1 other owner_.
+
+If you are transferring ownership of your organization to someone else, then you will need to invite the new member with the role `"Owner"`, and then once they've accpeted the invitation, you can then leave your organization.
```
Comment (saltcod) [inline apps/docs/pages/guides/platform/access-control.mdx:47]:
```suggestion
If you are transferring ownership of your organization to someone else, then you will need to invite the new member with the role `"Owner"`. Once they've accepted the invitation, you can then leave your organization.
```

File: apps/docs/pages/guides/platform/access-control.mdx [hunk 2]
```diff
@@ -37,6 +37,15 @@ You can invite your team members into your organizations to collaborate on proje
...
 </video>
 
+### Transferring ownership of an organization
+
+An organization can have multiple owners, so a Supabase organization can either have 1 owner, or several owners.
+
+If you would like to no longer be an owner of an organization you can simply click 'Leave team' in the members view (`https://app.supabase.com/org/<org-slug>/settings#team`) of your organization.
+However, you can only leave an organization when there is _at least 1 other owner_.
+
+If you are transferring ownership of your organization to someone else, then you will need to invite the new member with the role `"Owner"`. Once they've accepted the invitation, you can then leave your organization.
```
Comment (dannykng) [inline apps/docs/pages/guides/platform/access-control.mdx:47]:
```suggestion
Each Supabase organization can have one or more owners. If you no longer want be an owner of an organization, click **Leave team** in the members view (`https://app.supabase.com/org/<org-slug>/settings#team`) of your organization.
However, you can only leave an organization when there is _at least one other owner_.

If you are transferring ownership of your organization to someone else, you will need to invite the new member with the **Owner** role. You can leave the organization after they've accepted the invitation.
```

**Review by dannykng — APPROVED** [review]
✅ on docs

File: studio/components/interfaces/Organization/TeamSettings/MembersView.tsx [hunk 3]
```diff
@@ -126,13 +126,17 @@ const MembersView = () => {
                               <span className="flex rounded-full border-2 border-border-secondary-light p-2 dark:border-border-secondary-dark">
                                 <IconUser size={20} strokeWidth={2} />
                               </span>
-                            ) : (
+                            ) : !x?.username?.includes('@') ? (
```
Comment (joshenlim) [inline studio/components/interfaces/Organization/TeamSettings/MembersView.tsx:129]: is this a correct check though or was there an update in the /members endpoint? the username doesn't include the "@" symbol
image

File: studio/components/interfaces/Organization/TeamSettings/TeamSettings.tsx [hunk 4]
```diff
@@ -24,6 +24,11 @@ const TeamSettings = observer(() => {
   const [isLeaving, setIsLeaving] = useState(false)
   const canAddMembers = rolesAddable.length > 0
 
+  const canLeave = (isOwner && ui.isOwnerAndCanLeaveOrg(members)) || !isOwner
+
+  console.log(rolesAddable)
```
Comment (joshenlim) [inline studio/components/interfaces/Organization/TeamSettings/TeamSettings.tsx:29]: remove logs

File: studio/components/interfaces/Organization/TeamSettings/TeamSettings.tsx [hunk 5]
```diff
@@ -24,6 +24,11 @@ const TeamSettings = observer(() => {
   const [isLeaving, setIsLeaving] = useState(false)
   const canAddMembers = rolesAddable.length > 0
 
+  const canLeave = (isOwner && ui.isOwnerAndCanLeaveOrg(members)) || !isOwner
```
Comment (joshenlim) [inline studio/components/interfaces/Organization/TeamSettings/TeamSettings.tsx:27]: `(isOwner && ui.isOwnerAndCanLeaveOrg(members))`

i think the `isOwner` check here is unnecessary right? can just be
`const canLeave = (ui.isOwnerAndCanLeaveOrg(members)) || !isOwner`

File: studio/stores/UiStore.ts [hunk 6]
```diff
@@ -157,4 +168,29 @@ export default class UiStore implements IUiStore {
...
+    const membersWhoAreOwners = members.filter((member) => {
+      const [memberRoleId] = member.role_ids ?? []
+      const role = (roles.roles || []).find((role: Role) => role.id === memberRoleId)
+      // dangerous use of string here
+      // to do, refactor so that changing the role name does not fail this check
+      // if role name does change, then user will simply not be able to leave project as a 'owner' regardless
+      return role && role.name === 'Owner' && !member.invited_at
+    })
+
+    console.log('membersWhoAreOwners', membersWhoAreOwners)
```
Comment (joshenlim) [inline studio/stores/UiStore.ts:190]: remove logs

File: studio/stores/UiStore.ts [hunk 7]
```diff
@@ -157,4 +168,29 @@ export default class UiStore implements IUiStore {
...
+
+    return selectedOrg?.is_owner && membersWhoAreOwners && membersWhoAreOwners.length > 1
```
Comment (joshenlim) [inline studio/stores/UiStore.ts:192]: just `return selectedOrg?.is_owner && membersWhoAreOwners && membersWhoAreOwners.length > 1` is enough actually, don't need `? true : false`

**Comment by joshenlim** [comment]
@MildTomato i'm testing this out locally, but i don't see the option to leave the org as an owner
image

**Review by joshenlim — CHANGES_REQUESTED** [review]
left some comments on the code

**Comment by joshenlim** [comment]
as an improvement actually - could we disable the leave team button instead of hiding it? then add a tooltip on why the user can't leave the org (need at least 1 owner)

File: studio/components/interfaces/Organization/TeamSettings/MembersView.tsx (hunk 3 above)
Comment (MildTomato) [reply studio/components/interfaces/Organization/TeamSettings/MembersView.tsx:129]: anyone got better ideas?

I commented on the same thing in someone elses PR which was dealing with the same thing.
Can we check what auth provider the user is using?

**Review by joshenlim — APPROVED** [review]
pushed up the changes and fixes

**Pull request merged**